client/src/app/pages/migration-waves/migration-waves.tsx

```diff
@@ -105,16 +108,23 @@ export const MigrationWaves: React.FC = () => {
   const openCreateWaveModal = () => setWaveModalState("create");
 
   const [exportIssueModalOpen, setExportIssueModalOpen] = React.useState(false);
-
   const [applicationsToExport, setApplicationsToExport] = React.useState<Ref[]>(
     []
   );
 
   const [waveToManageModalState, setWaveToManageModalState] =
     React.useState<MigrationWave | null>(null);
-
   const closeWaveModal = () => setWaveModalState(null);
 
+  const [migrationWaveToDelete, setMigrationWaveToDelete] = React.useState<{
+    id: number;
+    name: string;
+  } | null>(null);
+
+  const [migrationWavesToDelete, setMigrationWavesToDelete] = React.useState<
+    number[] | null
+  >(null);
+
```

Comment:
I think we should avoid using useState here but I see why you did it for the confirm modal. Hmm..

Reply:
This is probably fine. Just wondering if there is a better way rather than cluttering our components up. Can revisit later.

Reply:
@ibolton336, 

We have to use state because we capture the item to delete which will be deleted after Confirmation Dialog.
Combining the state and flag was introduced by @mturley (https://github.com/konveyor/tackle2-ui/pull/711) and I think it helps simplifying.

Reply:
Yeah I understand that. But now we have duplicated some state here it appears.

Reply:
@ibolton336, hmmm, we have state for exporting, managing, editing and deleting, let me harmonize the names and align them.

Reply:
Ok. I think you could drop the duplicate wave / waves state and just use one slice of state for that. Not sure I follow on your other point.